solver/llbsolver/file/backend.go

```diff
@@ -335,14 +309,41 @@ func (fb *Backend) Copy(ctx context.Context, m1, m2, user, group fileoptypes.Mou
 	}
 	defer lm2.Unmount()
 
-	u, err := readUser(action.Owner, user, group)
+	u, err := fb.readUserWrapper(action.Owner, user, group)
 	if err != nil {
 		return err
 	}
 
 	return docopy(ctx, src, dest, action, u, mnt2.m.IdentityMapping())
 }
 
+func (fb *Backend) readUserWrapper(owner *pb.ChownOpt, user, group fileoptypes.Mount) (*copy.User, error) {
+	var userMountable, groupMountable snapshot.Mountable
+	if user != nil {
+		usr, ok := user.(*Mount)
```

Comment:
There is still something iffy about these type casts but looks like the previous code had a similar problem, so no update is needed with this PR.

Reply:
I agree. There are a few architectural boundaries being broken here which could do with some refactor, but it will require touching a lot of code. This pattern is present in a number of places and should probably be tackled at some point as part of a broader cleanup.

Thanks for the review and the merge!